bridges/FlaschenpostBridge.php

```diff
@@ -0,0 +1,290 @@
+<?php
+class FlaschenpostBridge extends BridgeAbstract {
+	const NAME = 'Flaschenpost Bridge';
+	const URI = 'https://www.flaschenpost.de/';
+	const DESCRIPTION = 'Aktuelle Angebote auf Flaschenpost.de';
+	const MAINTAINER = 'sal0max';
+	const CACHE_TIMEOUT = 3600; // 1 hour
+	const PARAMETERS = array(
+		array(
+			'zip-code' => array(
+				'name' => 'Postleitzahl',
+				'type' => 'text',
+				'required' => true,
+				'exampleValue' => '80333',
+				// https://stackoverflow.com/a/7926743/421140
+				'pattern' => '^(?!01000|99999)(0[1-9]\d{3}|[1-9]\d{4})$',
+			),
+			'water' => array(
+				'name' => 'Wasser',
+				'type' => 'checkbox',
+			),
+			'beer' => array(
+				'name' => 'Bier',
+				'type' => 'checkbox',
+			),
+			'lemonade' => array(
+				'name' => 'Limonade',
+				'type' => 'checkbox',
+			),
+			'juice' => array(
+				'name' => 'Saft & Schorle',
+				'type' => 'checkbox',
+			),
+			'wine' => array(
+				'name' => 'Wein & Mehr',
+				'type' => 'checkbox',
+			),
+			'liquor' => array(
+				'name' => 'Spirituosen',
+				'type' => 'checkbox',
+			),
+			'food' => array(
+				'name' => 'Lebensmittel',
+				'type' => 'checkbox',
+			),
+			'household' => array(
+				'name' => 'Haushalt',
+				'type' => 'checkbox',
+			),
+		)
+	);
+
+	public function getIcon() {
+		return 'https://image.flaschenpost.de/CI/fp-favicon.png';
+	}
+
+	public function getName(): string {
+		$categories = array();
+		if ($this->getInput('water'))
+			array_push($categories, 'Wasser');
+		if ($this->getInput('beer'))
+			array_push($categories, 'Bier');
+		if ($this->getInput('lemonade'))
+			array_push($categories, 'Limonade');
+		if ($this->getInput('juice'))
+			array_push($categories, 'Saft & Schorle');
+		if ($this->getInput('wine'))
+			array_push($categories, 'Wein & Mehr');
+		if ($this->getInput('liquor'))
+			array_push($categories, 'Spirituosen');
+		if ($this->getInput('food'))
+			array_push($categories, 'Lebensmittel');
+		if ($this->getInput('household'))
+			array_push($categories, 'Haushalt');
+		if (empty($categories)) {
+			return $this::NAME;
+		} else {
+			return $this::NAME . ' – ' . implode(', ', $categories);
+		}
+	}
+
+	public function collectData() {
+		// which categories to include
+		$urls = array();
+		if ($this->getInput('water'))
+			array_push($urls,
```

Comment:
Use `array_merge`.

Reply:
You have to help me here. I don't really see how array_merge could improve the code in this instance.

Instead of
```php
$urls = array();
if ($this->getInput('water'))
	array_push($urls,
		'wasser/spritzig',
		'wasser/medium',
		'wasser/still',
		'wasser/aromatisiert',
		'wasser/heilwasser',
		'wasser/bio-wasser',
		'wasser/gourmet'
	);
```
it had to be like this, right?

```php
$urls = array();
if ($this->getInput('water'))
	$urls = array_merge($urls,
		array('wasser/spritzig',
		'wasser/medium',
		'wasser/still',
		'wasser/aromatisiert',
		'wasser/heilwasser',
		'wasser/bio-wasser',
		'wasser/gourmet')
	);
```

Imho that's not good:
* Worse readability
* An additional array has to be created only to be garbage collected after the merge.